controllers/ovndbcluster_controller.go

```diff
@@ -467,28 +471,13 @@ func (r *OVNDBClusterReconciler) reconcileNormal(ctx context.Context, instance *
 
 			// Filter out headless services
 			if svc.Spec.ClusterIP != "None" {
-				// Test using hostname instead of ip for dbAddress connection
-				//serviceHostname := fmt.Sprintf("%s.%s.svc", svc.Name, svc.GetNamespace())
-				//dbAddress = append(dbAddress, fmt.Sprintf("tcp:%s:%d", serviceHostname, svc.Spec.Ports[0].Port))
-
-				internalDbAddress = append(internalDbAddress, fmt.Sprintf("tcp:%s:%d", svc.Spec.ClusterIP, svcPort))
-				raftAddress = append(raftAddress, fmt.Sprintf("tcp:%s:%d", svc.Spec.ClusterIP, svc.Spec.Ports[1].Port))
-			}
-		}
-
-		// External dbAddress if networkAttachment is used
-		if instance.Spec.NetworkAttachment != "" {
-			net := instance.Namespace + "/" + instance.Spec.NetworkAttachment
-			if netStat, ok := instance.Status.NetworkAttachments[net]; ok {
-				for _, instanceIP := range netStat {
-					dbAddress = append(dbAddress, fmt.Sprintf("tcp:%s:%d", instanceIP, svcPort))
-				}
+				internalDbAddress = append(internalDbAddress, fmt.Sprintf("tcp:%s.%s.svc.cluster.local:%d", svc.Name, svc.Namespace, svcPort))
```

Comment:
I don't know where .svc comes from, but the cluster.local comes from the CoreDNS configuration (which I think we're using the default one), you can get which cluster are you using with `oc describe dns.operator/default`:
```
[ospng@shark15 devsetup]$ oc describe dns.operator/default
Name:         default
Namespace:    
Labels:       <none>
Annotations:  <none>
API Version:  operator.openshift.io/v1
Kind:         DNS
Metadata:
  Creation Timestamp:  2023-11-22T01:33:08Z
  Finalizers:
    dns.operator.openshift.io/dns-controller
  Generation:        1
  Resource Version:  32583
  UID:               07fd2434-68c8-4108-9506-e49ad13fc011
Spec:
  Cache:
    Negative TTL:  0s
    Positive TTL:  0s
  Log Level:       Normal
  Node Placement:
  Operator Log Level:  Normal
  Upstream Resolvers:
    Policy:             Sequential
    Protocol Strategy:  
    Transport Config:
    Upstreams:
      Port:  53
      Type:  SystemResolvConf
Status:
  Cluster Domain:  cluster.local  # <== HERE
  Cluster IP:      10.217.4.10
  Conditions:
    Last Transition Time:  2024-01-02T16:31:30Z
```
Without more investigation I don't know how could we do it from OVN operator, any ideas?

Reply:
@stuggi any idea if there's a way to programmatically construct an in-cluster FQDN, with svc.cluster.local postfix, without assuming the default coredns configuration?

Reply:
.svc is a k8s thing [1] . services can be resolved using `service-name.namespace.svc`, pods as `pod-name.namespace.pod` . So we can expect .svc for services. cluster.local is the default internal cluster domain. I am not familiar with a way to customize that internal domain. we need to check with others. I have found a jira which says its node supported to change it after initial deployment.

btw, this domain is different from the ingress domain.

[1] https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/

Reply:
hrmm https://github.com/kubernetes/website/issues/25758

Reply:
Since this is a prerequisite for TLS support I think it would be better for now to just hardcode `cluster.local` as a common const. Can create a JIRA to follow-up later on how to fetch cluster_domain at runtime or get unbound to use the search domain from resolv.conf